var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -20,6 +20,7 @@ class Petsc(Package, CudaPackage, ROCmPackage):
 
     version('main', branch='main')
 
+    version("3.17.4", sha256="99c127486722a3ffd95a268b4ceb0976cbf217926c681a9631bd7246eab8cb2a")
```

Comment:
Is this from upstream?

Reply:
yeah..so I assume you would prefer the patch? can do that if we decide to do merge this.

Reply:
If upstream, this is fine. Then when I reset `builtin` to upstream, we don't lose anything.

Reply:
Ok!